### ISSUE TITLE
[Bug]: ILSpy decompilation is better than AssetRipper decompilation

### ISSUE THREAD

**Issue by aldelaro5** [AssetRipper/AssetRipper#542]
### Are you on the latest version of AssetRipper?

Yes, I'm on the latest release of AssetRipper.

### Which release are you using?

GUI Version Linux x64

### Which game did this occur on?

Bug Fables: The Everlasting Sapling

steamdb: https://steamdb.info/app/1082710/

### Which Unity version did this occur on?

2018.4.12f1

### Is the game Mono or IL2Cpp?

Mono

### Describe the issue.

If I do an export all with the C# version set to 4, the export script setting set to decompile and the content level set to 2, I get code that is worse than the same preview version of ILSpy that AssetRipper uses. Specifically, I get code that looks like this:

```cs
	[CompilerGenerated]
	private sealed class _003CGetPlayerWithMostHP_003Ec__AnonStorey43
```

This is actually supposed to be a lambda for a Linq expression. I can tell because if I use ILSpyCmd with the same version and the same C# version specified, I do not get this code. What I get is an inlining of the lambda directly. This results in code that is closer to be able to be compiled than what AssetRipper exports. There are other issues like these, but this is just an example exported from the BattleControl monoscript.

### Relevant log output

[AssetRipper.log](https://github.com/AssetRipper/AssetRipper/files/9562294/AssetRipper.log)

**Comment by ds5678** [comment]
This could be due to a couple reasons:
* The transforms in [this folder](https://github.com/AssetRipper/AssetRipper/tree/06e107f736818d5f132fd873a0d59ea408847f07/AssetRipper.Library/Exporters/Scripts/Transforms) modify the abstract syntax tree.
* ILSpy might resolve from loaded assemblies, which AssetRipper can't do because of publishing.
* AssetRipper is having resolve issues for some other reason.

**Comment by aldelaro5** [comment]
This is fixed as of preview 0.3.0.1.

Unity 2018.4.12f1, mono (.NET 2.0 Subset) at level 2 script export.
